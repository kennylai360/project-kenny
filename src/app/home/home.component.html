<div class="container flex-center">
    <div class="row pt-5 mt-3">
        <div class="col-md-12 mb-3">
            <div class="intro-info-content text-center">

                <h1 class="white-text display-3 mb-5 wow fadeInDownBig" data-wow-delay="0.5s">KENNY <a
                        class="indigo-text font-bold">LAI</a></h1>
                <h5 class="white-text font-up mb-5 mt-1 font-bold wow fadeInDownBig" data-wow-delay="0.3s">Frontend
                    developer & Photographer</h5>
                <a class="btn indigo lighten-2 btn-lg animated fadeInDownBig waves-light" mdbRippleRadius routerLink="/galleries">Portfolio</a>
                <a class="btn indigo lighten-1 btn-lg animated fadeInDownBig waves-light" mdbRippleRadius href="https://github.com/kennylai360">Github</a>
                <a class="btn indigo btn-lg animated fadeInDownBig waves-light" mdbRippleRadius>Profile</a>

            </div>
        </div>
    </div>
</div>
