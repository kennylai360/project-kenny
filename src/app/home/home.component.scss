h1 {
  color: white;
}

.full-bg-img {
  width: 50% !important;
  transform:  translate(50%,0%)
}
