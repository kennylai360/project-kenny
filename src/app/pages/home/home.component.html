<div class="home__info animate__animated animate__fadeInDown">
  <h1>Kenny Kin Kee Lai</h1>
  <hr />
  <h5>Frontend Developer & Photographer</h5>
  <hr />
</div>
<div class="home__links animate__animated animate__fadeInDown">
  <a data-qa="qa-resume-link" routerLink="/resume">Resume</a>
  <a data-qa="qa-photography-link" routerLink="/photography">Photography</a>
  <a
    data-qa="qa-github-link"
    href="https://www.github.com/kennylai360"
    target="_blank"
    rel="noreferrer noopener"
    >Github</a
  >
</div>
