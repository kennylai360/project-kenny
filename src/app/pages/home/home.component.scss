.white-h1, .white-h5, .home-a {
  color: white;
}

hr {
  background-color: white
}

.title-size-correction {
  width: 50% !important;
  transform:  translate(50%,0%)
}
