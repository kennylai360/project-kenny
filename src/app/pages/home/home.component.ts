import { Component, OnInit, ViewEncapsulation } from '@angular/core';
import { CounterFacade } from '../../state-management/counter/counter-facade';

@Component({
  selector: 'app-home',
  templateUrl: './home.component.html',
  styleUrls: ['./home.component.scss'],
  encapsulation: ViewEncapsulation.None
})
export class HomeComponent implements OnInit {

  constructor(private counterFacade:  CounterFacade) { }

  ngOnInit() {
  }

  public addOne() {
    this.counterFacade.incrementCounter();
  }

}
