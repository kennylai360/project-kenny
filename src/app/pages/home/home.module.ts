import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { GalleryCoverModule } from '../../components/gallery-cover/gallery-cover.module';
import { NgxPaginationModule } from 'ngx-pagination';
import { GalleriesComponent } from '../galleries/galleries.component';
import { HomeComponent } from './home.component';
import { RouterModule } from '@angular/router';

@NgModule({
  declarations: [
    HomeComponent
  ],
  imports: [
    CommonModule,
    RouterModule
  ],
  exports: [
    HomeComponent
  ],
  providers: []
})
export class HomeModule {
}
