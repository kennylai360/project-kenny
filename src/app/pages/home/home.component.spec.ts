import { TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { By } from '@angular/platform-browser';
import { HomeComponent } from './home.component';

describe('Home component', () => {
  let fixture;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [HomeComponent, RouterTestingModule],
    }).compileComponents();

    fixture = TestBed.createComponent(HomeComponent);
  });

  describe('Links and attributes', () => {
    it('should test the resume link will have routerLink to /resume', () => {
      const resumeRouterLink = fixture.debugElement
        .query(By.css('a[data-qa="qa-resume-link"]'))
        .nativeElement.getAttribute('routerLink');
      expect(resumeRouterLink).toEqual('/resume');
    });

    it('should test the photography link will have routerLink to /photography', () => {
      const photographyRouterLink = fixture.debugElement
        .query(By.css('a[data-qa="qa-photography-link"]'))
        .nativeElement.getAttribute('routerLink');
      expect(photographyRouterLink).toEqual('/photography');
    });

    it('should test the github link will have routerLink to the correct github url', () => {
      const githubLink = fixture.debugElement
        .query(By.css('a[data-qa="qa-github-link"]'))
        .nativeElement.getAttribute('href');
      expect(githubLink).toEqual('https://www.github.com/kennylai360');
    });

    it('should check that the github link has a rel attribute with values of noreferrer noopener', () => {
      const githubRelAttribute = fixture.debugElement
        .query(By.css('a[data-qa="qa-github-link"]'))
        .nativeElement.getAttribute('rel');
      expect(githubRelAttribute).toEqual('noreferrer noopener');
    });
  });

  it('should check the home UI should match snapshots', () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
