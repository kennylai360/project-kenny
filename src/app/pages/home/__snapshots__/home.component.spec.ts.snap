// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should check the home UI should match snapshots 1`] = `
<div
  id="root4"
>
  <div
    class="home__info animate__animated animate__fadeInDown"
  >
    <h1>
      Kenny Kin Kee Lai
    </h1>
    <hr />
    <h5>
      Frontend Developer & Photographer
    </h5>
    <hr />
  </div>
  <div
    class="home__links animate__animated animate__fadeInDown"
  >
    <a
      data-qa="qa-resume-link"
      routerlink="/resume"
    >
      Resume
    </a>
    <a
      data-qa="qa-photography-link"
      routerlink="/photography"
    >
      Photography
    </a>
    <a
      data-qa="qa-github-link"
      href="https://www.github.com/kennylai360"
      rel="noreferrer noopener"
      target="_blank"
    >
      Github
    </a>
  </div>
</div>
`;
